src/hotspot/share/gc/z/zAllocator.inline.hpp

```diff
@@ -27,14 +27,15 @@
 #include "gc/z/zAllocator.hpp"
 
 #include "gc/z/zAddress.inline.hpp"
+#include "gc/z/zPageAge.inline.hpp"
 #include "gc/z/zHeap.hpp"
 
 inline ZAllocatorEden* ZAllocator::eden() {
   return _eden;
 }
 
 inline ZAllocatorForRelocation* ZAllocator::relocation(ZPageAge page_age) {
-  return _relocation[static_cast<uint>(page_age) - 1];
+  return _relocation[untype(page_age) - 1];
```

Comment:
I wonder if we should have our own defined ZPageAge operators for `+` and `-` (or maybe inc/dec) and we could add verification that we don't fall out of the range?
```suggestion
  return _relocation[untype(page_age - 1)];
```

Reply:
I agree. That also allows us to do something similar to how the offset types are handled in zAddress.inline.hpp, to check if the size being added/substracted is within the underlying type.